src/Compilers/CSharp/Portable/Syntax/SyntaxFactory.cs

```diff
@@ -1718,7 +1718,7 @@ public static TypeSyntax ParseTypeName(string text, int offset = 0, ParseOptions
             {
                 var node = parser.ParseTypeName();
                 if (consumeFullText) node = parser.ConsumeUnexpectedTokens(node);
-                return (TypeSyntax)node.CreateRed();
+                return CreateRed<TypeSyntax>(node, lexer.Options);
```

Comment:
Are there more use sites that should be modified in a similar way? e.g. ParseName on line 1690.

Reply:
Other `Parse[NodeName]` don't accept parse options, so initially I didn't touch them. But I believe for the purpose of correctness we should use this helper in them as well. The downside is that we will eagerly allocate `SyntaxTree` object on each parse no matter whether it is later used or not